### ISSUE TITLE
IndexError in resolve_strategy_inline_outputs

### ISSUE THREAD

**Issue by nucflash** [jupyter/nbdime#525]
Hi, 

My `.gitconfig` looks like this:
```
[diff "jupyternotebook"]
        command = git-nbdiffdriver diff
[merge "jupyternotebook"]
        driver = git-nbmergedriver merge %O %A %B %L %P
        name = jupyter notebook merge driver
[difftool "nbdime"]
        cmd = git-nbdifftool diff \"$LOCAL\" \"$REMOTE\" \"$BASE\"
[difftool]
        prompt = false
[mergetool "nbdime"]
        cmd = git-nbmergetool merge \"$BASE\" \"$LOCAL\" \"$REMOTE\" \"$MERGED\"
[mergetool]
        prompt = true
```

When I run `git merge <branch>`, I get this error:
```
[W strategies:469] Recording unresolved conflicts in /cells/3/metadata/nbdime-conflicts.
Traceback (most recent call last):
  File "/app/python/bin/git-nbmergedriver", line 8, in <module>
    sys.exit(main())
  File "/app/python/lib/python3.6/site-packages/nbdime/vcs/git/mergedriver.py", line 115, in main
    return nbmergeapp.main_merge(opts)
  File "/app/python/lib/python3.6/site-packages/nbdime/nbmergeapp.py", line 51, in main_merge
    merged, decisions = merge_notebooks(b, l, r, args)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/notebooks.py", line 182, in merge_notebooks
    decisions = decide_notebook_merge(base, local, remote, args)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/notebooks.py", line 156, in decide_notebook_merge
    strategies)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/generic.py", line 737, in decide_merge_with_diff
    path, parent_decisions, strategies)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/generic.py", line 712, in _merge
    path, decisions, strategies)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/generic.py", line 266, in _merge_dicts
    subdecisions = _merge(bv, ld.diff, rd.diff, item_path, decisions, strategies)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/generic.py", line 716, in _merge
    path, decisions, strategies)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/generic.py", line 544, in _merge_lists
    item_path, decisions, strategies)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/generic.py", line 712, in _merge
    path, decisions, strategies)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/generic.py", line 266, in _merge_dicts
    subdecisions = _merge(bv, ld.diff, rd.diff, item_path, decisions, strategies)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/generic.py", line 716, in _merge
    path, decisions, strategies)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/generic.py", line 635, in _merge_lists
    resolve_conflicted_decisions_list(path, base, decisions, list_strategy)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/strategies.py", line 641, in resolve_conflicted_decisions_list
    resolve_strategy_inline_outputs(path, base, decisions)
  File "/app/python/lib/python3.6/site-packages/nbdime/merging/strategies.py", line 432, in resolve_strategy_inline_outputs
    outputs[key], local_diff, remote_diff)
IndexError: list index out of range
```

I have tried different merging strategies to no avail. Unfortunately I cannot share my notebooks. Any thoughts on what can it be going wrong?

**Comment by vidartf** [comment]
Thanks for the report!

Given the stack trace, it seems to be failing when trying to resolve the merge on outputs using the inline strategy. There are two possible workarounds, both which would add a flag to the `driver = git-nbmergedriver merge %O %A %B %L %P` line:
- The first thing to try might be to add `--output-strategy=<value>` where `<value>` is one of:
  - "use-base",         # Keep base value in case of conflict
  - "use-local",        # Use local value in case of conflict
  - "use-remote",       # Use remote value in case of conflict
  - "remove",     # Remove conflicting outputs
  - "clear-all",  # Clear all outputs
  I'd be interesting in hearing which ones work and which doesn't (if any) for you case, since you are not able to share the repro.
- If none of the above work (possible if the bug is further up the chain), and if you don't care about the outputs you can add the `--ignore-outputs` flag (it will disregard outputs both for comparisons of cells, and as a source of conflicts, but might not give you the "correct' outputs).

To truly debug this, a reproducible example would go a long way. One way you might be able to share such an example without revealing your notebooks is this:
- Extract the base / local / remote versions of the notebook using git
- Try to change the local and remote step by step to become progressively more similar to the base until you can isolate only the changes needed for this error to occur.
- You can also edit these notebook to strip anything that would be considered sensitive.
- Then, either share the "anonymized" notebooks, or describe which differences seem crucial to cause the error to appear.

Hope this helps!

**Comment by vidartf** [comment]
Closing due to lack of information in order to reproduce. If anyone else has this problem and are able to share some more information, please comment, and we can reopen the issue.

**Comment by pew-pew** [comment]
Hello, I was able to reproduce the issue. In the [attached archive](https://github.com/jupyter/nbdime/files/6257864/nbdime_merge_bug.zip) there are three notebooks - `o.ipynb` (common base), `a.ipynb` (my branch), `b.ipynb` (remote branch).

Trying to merge `a` and `b` gives the same error:
```shell
$ git-nbmergedriver --version
git-nbmergedriver 2.1.0
$ git-nbmergedriver merge o.ipynb a.ipynb b.ipynb 7 out.ipynb 
Traceback (most recent call last):
  File "/home/seal/.local/bin/git-nbmergedriver", line 8, in <module>
    sys.exit(main())
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/vcs/git/mergedriver.py", line 115, in main
    return nbmergeapp.main_merge(opts)
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/nbmergeapp.py", line 52, in main_merge
    merged, decisions = merge_notebooks(b, l, r, args)
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/notebooks.py", line 182, in merge_notebooks
    decisions = decide_notebook_merge(base, local, remote, args)
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/notebooks.py", line 153, in decide_notebook_merge
    decisions = decide_merge_with_diff(
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/generic.py", line 735, in decide_merge_with_diff
    decisions = _merge(
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/generic.py", line 710, in _merge
    return _merge_dicts(
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/generic.py", line 266, in _merge_dicts
    subdecisions = _merge(bv, ld.diff, rd.diff, item_path, decisions, strategies)
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/generic.py", line 714, in _merge
    return _merge_lists(
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/generic.py", line 542, in _merge_lists
    subdecisions = _merge(
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/generic.py", line 710, in _merge
    return _merge_dicts(
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/generic.py", line 266, in _merge_dicts
    subdecisions = _merge(bv, ld.diff, rd.diff, item_path, decisions, strategies)
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/generic.py", line 714, in _merge
    return _merge_lists(
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/generic.py", line 635, in _merge_lists
    resolve_conflicted_decisions_list(path, base, decisions, list_strategy)
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/strategies.py", line 641, in resolve_conflicted_decisions_list
    resolve_strategy_inline_outputs(path, base, decisions)
  File "/home/seal/.local/lib/python3.9/site-packages/nbdime/merging/strategies.py", line 432, in resolve_strategy_inline_outputs
    outputs[key], local_diff, remote_diff)
IndexError: list index out of range
```

The only difference is the `output` section of a single cell (apart from `execution_count` fields):

`o.ipynb`:
```json
"outputs": []
```

`a.ipynb`:
```json
"outputs": [
    {
        "name": "stdout",
        "output_type": "stream",
        "text": [
            "AAA\n",
            "BBB\n"
        ]
    }
]
```

`b.ipynb`:
```json
"outputs": [
    {
        "name": "stdout",
        "output_type": "stream",
        "text": [
            "AAA\n"
        ]
    }
]
```

**Comment by pew-pew** [comment]
This issue was already fixed with commit 1184f78cd13cbf9635a51239965fff325354742a , it's sufficient to do `pip install "nbdime>=3.0.0b1"`
